packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -516,18 +451,14 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
             // Successful call, redirect future calls to getVersion only!
             this.firstVersionCall = false;
 
-            const odspSnapshot: IOdspSnapshot = odspSnapshotCacheValue.snapshot;
             this._snapshotSequenceNumber = odspSnapshotCacheValue.sequenceNumber;
-
-            const { trees, blobs, ops } = odspSnapshot;
+            const { snapshotTree, blobs, ops } = odspSnapshotCacheValue;
             // id should be undefined in case of just ops in snapshot.
             let id: string | undefined;
-            if (trees) {
-                this.initCommitCache(trees);
-                // versionId is the id of the first tree
-                if (trees.length > 0) {
-                    id = trees[0].id;
-                }
+            if (snapshotTree) {
+                id = snapshotTree.id;
+                assert(id !== undefined, "Root tree should contain the id");
+                this.initCommitCache(new Map([[id, snapshotTree]]));
```

Comment:
The name and typing no longer match. Also it always received exactly one entry.
So I'd change it to something like setRootTree(id, tree);